### ISSUE TITLE
ne_vec macro expands expr metavariables in unsafe block, allowing unsafe code without unsafe blocks

### ISSUE THREAD

**Issue by y21** [yihuang/non-empty-vec#18]
I noticed that the `ne_vec` macro expands its metavariables in an unsafe block:
https://github.com/yihuang/non-empty-vec/blob/8b15d1f87445560f9afb5d7b5e0234686d0691ee/src/lib.rs#L363-L365

This means that at macro call site, the user can write unsafe code inside of the `ne_vec!` macro without an unsafe block, because when expanded the element is inside of a hidden unsafe block introduced by the macro:
```rs
let ve = ne_vec![*std::ptr::null::<u8>()];
// becomes
let ve = unsafe { non_empty_vec::NonEmpty::new_unchecked(vec![*std::ptr::null::<u8>()]) };
```

We can probably fix this by just moving the metavariables outside of the unsafe block and store them in a temporary variable.
Something like
```rs
    ($($x:expr),+ $(,)?) => {{
        let ve = vec![$($x),+];
        unsafe { $crate::NonEmpty::new_unchecked(ve) }
    }};
```

On an unrelated note, we should probably also use an absolute path to `vec!` like we are doing for `::std::compile_error`?

**Comment by joseph-gio** [comment]
Nice catch! This one was my bad, so thanks!

**Comment by yihuang** [comment]
fixed by #13
